cmd/crates/soroban-spec-typescript/src/project_template/src/assembled-tx.ts

```diff
@@ -0,0 +1,584 @@
+        retval: this.simulationResult.retval.toXDR('base64'),
+      },
+      simulationTransactionData: this.simulationTransactionData.toXDR('base64'),
+    })
+  }
+
+  static fromJSON<T>(
+    options: Omit<AssembledTransactionOptions<T>, 'args'>,
+    { tx, simulationResult, simulationTransactionData }:
+    {
+      tx: XDR_BASE64,
+      simulationResult: {
+        auth: XDR_BASE64[],
+        retval: XDR_BASE64,
+      },
+      simulationTransactionData: XDR_BASE64,
+    }
+  ): AssembledTransaction<T> {
+    const txn = new AssembledTransaction(options)
+    txn.tx = TransactionBuilder.fromXDR(tx, options.networkPassphrase) as Tx
+    txn.simulationResult = {
+      auth: simulationResult.auth.map(a => xdr.SorobanAuthorizationEntry.fromXDR(a, 'base64')),
+      retval: xdr.ScVal.fromXDR(simulationResult.retval, 'base64'),
+    }
+    txn.simulationTransactionData = xdr.SorobanTransactionData.fromXDR(simulationTransactionData, 'base64')
+    return txn
+  }
+
+  private constructor(public options: AssembledTransactionOptions<T>) {
+    this.server = new Server(this.options.rpcUrl, {
+      allowHttp: this.options.rpcUrl.startsWith("http://"),
+    });
+  }
+
+  static async fromSimulation<T>(options: AssembledTransactionOptions<T>): Promise<AssembledTransaction<T>> {
+    const tx = new AssembledTransaction(options)
+    const contract = new Contract(options.contractId);
+
+    tx.tx = new TransactionBuilder(await tx.getAccount(), {
+      fee: options.fee?.toString(10) ?? '100',
+      networkPassphrase: options.networkPassphrase,
+    })
+      .addOperation(contract.call(options.method, ...(options.args ?? [])))
+      .setTimeout(TimeoutInfinite)
+      .build();
+
+    return await tx.simulate()
+  }
+
+  simulate = async (): Promise<this> => {
+    if (!this.tx) throw new Error('Transaction has not yet been assembled')
+    const simulation = await this.server.simulateTransaction(this.tx);
+
+    if (!simulation) {
+      throw new Error('Transaction has not yet been simulated')
+    }
+
+    if (SorobanRpc.isSimulationError(simulation)) {
+      throw new Error(`Transaction simulation failed: "${simulation.error}"`)
+    }
+
+    if (SorobanRpc.isSimulationRestore(simulation)) {
+      throw new ExpiredStateError(`You need to restore some contract state before you can invoke this method. ${JSON.stringify(simulation, null, 2)}`)
+    }
+
+    if (!simulation.result) {
+      throw new Error(`Expected an invocation simulation, but got no 'result' field. Simulation: ${JSON.stringify(simulation, null, 2)}`)
+    }
+
+    this.simulationResult = simulation.result
+    this.simulationTransactionData = simulation.transactionData.build()
+
+    this.tx = assembleTransaction(
+      this.tx,
+      this.options.networkPassphrase,
+      simulation
+    ).build()
+
+    return this
+  }
+
+  getWallet = async (): Promise<Wallet> => {
+    return this.options.wallet ?? (await import("@stellar/freighter-api"))
```

Comment:
small thing here, but I think we'll need to add `.default` to the end of this like we did [here](https://github.com/stellar/soroban-tools/blob/733ce062532c45ff96e690f135f6c1193a6d14d3/cmd/crates/soroban-spec-typescript/src/project_template/src/invoke.ts#L86)